### PR TITLE
Graphclust visualizations of the clusters

### PR DESCRIPTION
**Pull request by mmiladi**
Cool update, we now have visualizations of the cluster alignments and their secondary structures.

### CODE REVIEW

**Comment by mmiladi** [comment]
@bgruening The invoked perl script, uses `convert` command to convert .ps files to .png. 
The dependency for imagemagick is missing.

**Comment by bgruening** [comment]
We should add a test for this, isn't it?

**Comment by bgruening** [comment]
@mmiladi use graphicsmagick this works as well.

**Comment by mmiladi** [comment]
Yes test is needed.

**Comment by mmiladi** [comment]
You mean this one? https://anaconda.org/bioconda/graphicsmagick/
Not certain if it works.:
```
$ conda install -c bioconda graphicsmagick
$ gm convert cluster.top5.aln.ps ff.png
gm convert: Unable to access configuration file (delegates.mgk) [No such file or directory].

```

**Comment by bgruening** [comment]
just convert?

**Comment by mmiladi** [comment]
Nee. `convert` calls imagesmagick from /usr/bin.
 graphicsmagick is `gm convert` 
http://www.graphicsmagick.org/convert.html

**Comment by mmiladi** [comment]
Sorry but I think this is the same case for here :-P
https://github.com/bgruening/galaxytools/blob/4ecb336ec664bd30258dd0887db6da8459e9a938/tools/image_processing/graphicsmagick/montage.xml

**Comment by bgruening** [comment]
You are not using it in any environment, please do so and it will work :)

**Comment by mmiladi** [comment]
Nope :-D. 
```
$ source activate graphicmagick
(graphicmagick) rock:> $ conda install -c bioconda graphicsmagick
(graphicmagick) rock:> $ gm convert cluster.top5.aln.ps ff.png
gm convert: Unable to access configuration file (delegates.mgk) [No such file or directory].

```

Look here you call  `convert` not `gm convert` https://github.com/bgruening/galaxytools/blob/master/tools/image_processing/graphicsmagick/convert.xml

**Comment by mmiladi** [comment]
@eteriSokhoyan Could you please test both old and new outputs? The png outputs needs new test case, also please verify cluster outputs.
Please test the updated wrapper and code files: https://github.com/mmiladi/galaxytools/blob/graphclust-update/tools/GraphClust/CollectResults/glob_report.xml
https://github.com/BackofenLab/GraphClust/blob/galaxy_wrappers/galaxy_wrappers/CollectResults/gc_res.pl

File: tools/GraphClust/CollectResults/glob_report.xml [hunk 1]
```diff
@@ -3,6 +3,11 @@
 		<requirement type="package" version="0.1">graphclust-wrappers</requirement>
 		<requirement type="package" version='0.5'>perl-array-utils</requirement>
 		<requirement type="package" version='0.18.1'>scikit-learn</requirement>
+		<requirement type="package" version='1.8.10'>locarna</requirement>
+		<requirement type="package" version='2.1'>rnaz</requirement>
```
Comment (bgruening) [inline tools/GraphClust/CollectResults/glob_report.xml]: Why was all this not needed before?

Comment (bgruening) [inline tools/GraphClust/CollectResults/glob_report.xml]: Test does pass, if you can answer me this question, I'm ready to merge :)

**Comment by eteriSokhoyan** [comment]
Hey,

We didn't need this part before, cz before we didn't have the
visualization) and for making the visualization we need the locarna and the
other things.


On Jan 13, 2017 21:58, "Björn Grüning" <notifications@github.com> wrote:

*@bgruening* commented on this pull request.
------------------------------

In tools/GraphClust/CollectResults/glob_report.xml
<https://github.com/bgruening/galaxytools/pull/486>:

> @@ -3,6 +3,11 @@
 		<requirement type="package" version="0.1">graphclust-wrappers</requirement>
 		<requirement type="package" version='0.5'>perl-array-utils</requirement>
 		<requirement type="package" version='0.18.1'>scikit-learn</requirement>
+		<requirement type="package" version='1.8.10'>locarna</requirement>
+		<requirement type="package" version='2.1'>rnaz</requirement>

Test does pass, if you can answer me this question, I'm ready to merge :)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/bgruening/galaxytools/pull/486>, or mute the thread

**Comment by mmiladi** [comment]
@bgruening @eteriSokhoyan Danke!

**Pull request merged**